downstairs/src/region.rs

```diff
@@ -965,6 +1106,162 @@ impl Region {
         Ok(())
     }
 
+    /**
+     * Repair an extent from another downstairs
+     *
+     * We need to repair an extent in such a way that an interruption
+     * at any time can be recovered from.
+     *
+     * Let us assume we are repairing extent 012
+     *  1. Make new 012.copy dir  (extent name plus: .copy)
+     *  2. Get all extent files from source side, put in 012.copy directory
+     *  3. fsync files we just downloaded
+     *  4. Rename 012.copy dir to 012.replace dir
+     *  5. fsync extent directory ( 00/000/ where the extent files live)
+     *  6. Replace current extent 012 files with copied files of same name
+     *    from 012.replace dir
+     *  7. Remove any files in extent dir that don't exist in replacing dir
```

Comment:
Yes (and I'll update the comment). 
The replacement extent set of files could be:
```
012
012.db
```
And the current extent (that is old, that we are repairing) has:
```
012
012.db
012.db-wal
012.db-shm
```
We want to (after replacing `012` and `012.db`) Remove `012.db-shm` and `012.db-wal`, otherwise
we risk corrupting our extent.